### PR TITLE
Added TF-Lite-compatible feature extractor and model exporter for YAMNet

### PR DESCRIPTION
**Pull request by plakal**
- Added a TF-Lite compatible feature extractor. With the latest TF-Lite,
  that involves a DFT-multiplication replacement for tf.abs(tf.signal.stft())
  and not a lot else. Note that TF-Lite now allows variable-length inputs.
- Added a YAMNet exporter that produces TF2 SavedModels, TF-Lite models,
  and TF-JS models.
- Cleanups: switched hyperparameters to a dataclass, got rid of
  some lingering cruft in yamnet_test.

# Description

> :memo: Please include a summary of the change. 
>  
> * Please also include relevant motivation and context.  
> * List any dependencies that are required for this change.  

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Tests

> :memo: Please describe the tests that you ran to verify your changes.
>  
> * Provide instructions so we can reproduce.  
> * Please also list any relevant details for your test configuration.  

**Test Configuration**:

Ran VGGish exporter, YAMNet {test, inference demo, exporter}

## Checklist

- [x] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [x] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [x] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [x] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: research/audioset/yamnet/export.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+  (assumed to be mono 16 kHz samples in the [-1, +1] range)
+and return as output:
+- a 2-d float32 Tensor of shape [num_frames, num_classes] containing
+  predicted class scores for each frame of audio extracted from the input.
+- a 2-d float32 Tensor of shape [num_frames, embedding_size] containing
+  embeddings of each frame of audio.
+- a 2-d float32 Tensor of shape [num_spectrogram_frames, num_mel_bins]
+  containing the log mel spectrogram of the entire waveform.
+The SavedModels will also contain (as an asset) a class map CSV file that maps
+class indices to AudioSet class names and Freebase MIDs.
```
Comment (dpwe) [inline research/audioset/yamnet/export.py:14]: Not all of them are Freebase-canonical, but close enough I guess.

File: research/audioset/yamnet/export.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+  print('\n=====\n{} | {}\n=====\n'.format(time.asctime(), msg), flush=True)
+
+
+class YAMNet(tf.Module):
+  "''A TF2 Module wrapper around YAMNet."""
+  def __init__(self, weights_path, params):
+    super().__init__()
+    self._yamnet = yamnet.yamnet_frames_model(params)
+    self._yamnet.load_weights(weights_path)
+    self._class_map_asset = tf.saved_model.Asset('yamnet_class_map.csv')
```
Comment (dpwe) [inline research/audioset/yamnet/export.py:51]: Do we ever test/demonstrate using this Asset?  It looks like check_model reads the one in the local directory.

File: research/audioset/yamnet/params.py [hunk 3]
```diff
@@ -15,25 +15,37 @@
...
-STFT_HOP_SECONDS = 0.010
-MEL_BANDS = 64
-MEL_MIN_HZ = 125
-MEL_MAX_HZ = 7500
-LOG_OFFSET = 0.001
-PATCH_WINDOW_SECONDS = 0.96
-PATCH_HOP_SECONDS = 0.48
+@dataclass(frozen=True)  # Instances of this class are immutable.
+class Params:
+  sample_rate: int = 16000
```
Comment (dpwe) [inline research/audioset/yamnet/params.py:26]: Sample rate isn't intrinsically integral, although I agree all the ones we're going to encounter are.

File: research/audioset/yamnet/params.py [hunk 4]
```diff
@@ -15,25 +15,37 @@
...
+  stft_window_seconds: float = 0.025
+  stft_hop_seconds: float = 0.010
+  mel_bands: int = 64
+  mel_min_hz: int = 125
+  mel_max_hz: int = 7500
```
Comment (dpwe) [inline research/audioset/yamnet/params.py:31]: These definitely aren't integers.

File: research/audioset/yamnet/params.py (hunk 3 above)
Comment (plakal) [reply research/audioset/yamnet/params.py:26]: switched to float

File: research/audioset/yamnet/params.py (hunk 4 above)
Comment (plakal) [reply research/audioset/yamnet/params.py:31]: switched to float

File: research/audioset/yamnet/export.py (hunk 2 above)
Comment (plakal) [reply research/audioset/yamnet/export.py:51]: Good point, done.

**Pull request merged**